### ISSUE TITLE
Failure to build master on macos

### ISSUE THREAD

**Issue by aardvarkk** [mapnik/mapnik#4051]
I'm having a failure building mapnik on macos. The failure is triggered when building the sqlite datasource, and specifically it runs into an undeclared identifier for `sqlite3_enable_load_extension` and `sqlite3_load_extension`.

I have noticed that the dependency checking from scons seems to think my sqlite installation is fine (it was installed via brew), but perhaps the loaded header file is somehow incorrect and doesn't include the required declarations?

Here's the full output:

```
c++ -o plugins/input/sqlite/sqlite_datasource.os -c -std=c++14 -DU_USING_ICU_NAMESPACE=0 -stdlib=libc++ -fvisibility=hidden -fvisibility-inlines-hidden -Wall -ftemplate-depth-300 -Wsign-compare -O3 -fPIC -DMAPNIK_MEMORY_MAPPED_FILE -DMAPNIK_HAS_DLCFN -DBIGINT -DHAVE_JPEG -DHAVE_PNG -DHAVE_WEBP -DHAVE_TIFF -DDARWIN -DMAPNIK_THREADSAFE -DBOOST_SPIRIT_NO_PREDEFINED_TERMINALS=1 -DBOOST_PHOENIX_NO_PREDEFINED_TERMINALS=1 -DBOOST_SPIRIT_USE_PHOENIX_V3=1 -DNDEBUG -Ideps -Ideps/mapbox/polylabel/include -Ideps/mapbox/protozero/include -Ideps/mapbox/geometry/include -Ideps/mapbox/variant/include -Ideps/agg/include -Iinclude -I/usr/local/Cellar/postgresql/11.2_1/include -I/usr/local/Cellar/gdal/2.4.1/include -I/usr/local/opt/freetype/include/freetype2 -I/usr/local/opt/icu4c/include -I/usr/local/include -I/usr/include plugins/input/sqlite/sqlite_datasource.cpp
In file included from plugins/input/sqlite/sqlite_datasource.cpp:23:
In file included from plugins/input/sqlite/sqlite_datasource.hpp:46:
plugins/input/sqlite/sqlite_connection.hpp:171:9: error: use of undeclared identifier 'sqlite3_enable_load_extension'
        sqlite3_enable_load_extension(db_, 1);
        ^
plugins/input/sqlite/sqlite_connection.hpp:172:22: error: use of undeclared identifier 'sqlite3_load_extension'
        int result = sqlite3_load_extension(db_, ext_path.c_str(), 0 , 0);
                     ^
2 errors generated.
scons: *** [plugins/input/sqlite/sqlite_datasource.os] Error 1
scons: building terminated because of errors.
make: *** [mapnik] Error 2
```

**Comment by aardvarkk** [comment]
I believe I've solved the problem.

macos was pulling in the `sqlite3.h` header from its platform includes (`/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/sqlite3.h`), and that version didn't support the sqlite3 extensions. 

My brew version did, which was what confused me so much. I had to run the install with the following arguments to get the offending files to compile:

```
./configure SQLITE_INCLUDES=/usr/local/opt/sqlite3/include
make
```
